fix(transformer/module_runner): hoist export

hi-ogawa opened the pull request:
Aligning with https://github.com/vitejs/vite/pull/18983

I manually verified the test case output matches with the ones in Vite repo. 

~I add a new TODO about "export default" getter, but this difference existed already. I plan to fix this separately.~ (EDIT: It turned out Vite's transform is the wrong one https://github.com/vitejs/vite/issues/19834.)

hi-ogawa commented:
* **#10347** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10347?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10347?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10347).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented on crates/oxc_transformer/src/plugins/module_runner_transform.rs:
```diff
@@ -168,18 +171,10 @@ impl<'a> ModuleRunnerTransform<'a> {
-            }
-        }
+        new_stmts.splice(0..0, hoist_imports);
+        new_stmts.splice(0..0, hoist_exports);
```
```suggestion
        new_stmts.splice(0..0, hoist_exports.into_iter().chain(hoist_imports);
```

This looks like it would work better - What do you think?

hi-ogawa replied:
It works! I don't understand the perf difference though :smile:

Dunqing commented:
### Merge activity

* **Apr 13, 10:36 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

hi-ogawa committed:
fix(transformer/module_runner): hoist export (#10347)

Aligning with https://github.com/vitejs/vite/pull/18983

I manually verified the test case output matches with the ones in Vite repo.

~I add a new TODO about "export default" getter, but this difference existed already. I plan to fix this separately.~ (EDIT: It turned out Vite's transform is the wrong one https://github.com/vitejs/vite/issues/19834.)
